src/riscv/lib/src/machine_state.rs

```diff
@@ -647,7 +648,8 @@ mod tests {
 
         let state_cell = std::cell::RefCell::new(state);
 
-        proptest!(|(
+        // Lower the number of iterations because a single one takes around 7 seconds.
+        proptest!(ProptestConfig::with_cases(10), |(
```

Comment:
I don't think it is ok to degrade the test quality for all backends.

Reply:
Found out the culprit: It's the `.reset()` method which makes the proof backend very slow. Even with `10` iterations, the CI is timing out, I'm trying with `8` now.

I managed to modify a bit the `backend_test!` macro to allow per-backend customization of the iterations

[diff](https://github.com/tezos/riscv-pvm/compare/1faabd2aec1f3afcf58fd12cff2015bf5ff26aac..ad0dbc16fabca6ebeb0d92ca696e4c33920889ea)

Reply:
Is there a way to improve the `reset()` method without needing to tweak the test config? Perhaps there is a way to improve the tests as well?